lib/ansible/modules/network/netscaler/netscaler_server.py

```diff
@@ -0,0 +1,386 @@
+
+extends_documentation_fragment: netscaler
+requirements:
+    - nitro python sdk
+'''
+
+EXAMPLES = '''
+- name: Setup server
+  delegate_to: localhost
+  netscaler_server:
+      nsip: 172.18.0.2
+      nitro_user: nsroot
+      nitro_pass: nsroot
+
+      state: present
+
+      name: server-1
+      ipaddress: 192.168.1.1
+'''
+
+RETURN = '''
+loglines:
+    description: list of logged messages by the module
+    returned: always
+    type: list
+    sample: ['message 1', 'message 2']
+
+msg:
+    description: Message detailing the failure reason
+    returned: failure
+    type: str
+    sample: "Action does not exist"
+
+diff:
+    description: List of differences between the actual configured object and the configuration specified in the module
+    returned: failure
+    type: dict
+    sample: { 'targetlbvserver': 'difference. ours: (str) server1 other: (str) server2' }
+'''
+
+from ansible.module_utils.basic import AnsibleModule
+
+from ansible.module_utils.netscaler import ConfigProxy, get_nitro_client, netscaler_common_arguments, log, loglines, get_immutables_intersection
+
+try:
+    from nssrc.com.citrix.netscaler.nitro.resource.config.basic.server import server
+    from nssrc.com.citrix.netscaler.nitro.exception.nitro_exception import nitro_exception
+    PYTHON_SDK_IMPORTED = True
+except ImportError as e:
+    PYTHON_SDK_IMPORTED = False
+
+
+def server_exists(client, module):
+    log('Checking if server exists')
+    if server.count_filtered(client, 'name:%s' % module.params['name']) > 0:
+        return True
+    else:
+        return False
+
+
+def server_identical(client, module, server_proxy):
+    log('Checking if configured server is identical')
+    if server.count_filtered(client, 'name:%s' % module.params['name']) == 0:
+        return False
+    server_list = server.get_filtered(client, 'name:%s' % module.params['name'])
+    if server_proxy.has_equal_attributes(server_list[0]):
+        return True
+    else:
+        return False
+
+
+def diff_list(client, module, server_proxy):
+    ret_val = server_proxy.diff_object(server.get_filtered(client, 'name:%s' % module.params['name'])[0]),
+    return ret_val[0]
+
+
+def do_state_change(client, module, server_proxy):
+    if module.params['disabled']:
+        log('Disabling server')
+        result = server.disable(client, server_proxy.actual)
+    else:
+        log('Enabling server')
+        result = server.enable(client, server_proxy.actual)
+    return result
+
+
+def main():
+
+    module_specific_arguments = dict(
+        name=dict(type='str'),
+        ipaddress=dict(type='str'),
+        domain=dict(type='str'),
+        translationip=dict(type='str'),
+        translationmask=dict(type='str'),
+        domainresolveretry=dict(type='int'),
+        ipv6address=dict(
+            type='bool',
+            default=False
+        ),
+        comment=dict(type='str'),
+        td=dict(type='float'),
+    )
+
+    hand_inserted_arguments = dict(
```

Comment:
The modules are initially auto generated by parsing the NITRO objects and producing the initial boilerplate code. Afterwards the manual changes take place.

To distinguish between the arguments that were auto generated and the ones added manually I have 2 dictionaries. ``module_specific_arguments`` is the auto generated dict and ``hand_inserted_arguments`` contains the manually added arguments.